### ISSUE TITLE
Weight Decay may matter more than muP for Learning Rate Transfer in Practice

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#448]
## Links

https://arxiv.org/abs/2510.19093

## Authors

[Atli Kosson](https://arxiv.org/search/cs?searchtype=author&query=Kosson,+A), [Jeremy Welborn](https://arxiv.org/search/cs?searchtype=author&query=Welborn,+J), [Yang Liu](https://arxiv.org/search/cs?searchtype=author&query=Liu,+Y), [Martin Jaggi](https://arxiv.org/search/cs?searchtype=author&query=Jaggi,+M), [Xi Chen](https://arxiv.org/search/cs?searchtype=author&query=Chen,+X)

### 3行要約

- 本論文は、Maximal Update Parameterization (µP) に基づく学習率スケーリングの実用的限界を再検討し、実際には独立した Weight Decay (WD) が学習率転移の主要因であることを示す。  
- µP の理論的前提であるアラインメント仮定（入力と勾配の幾何学的整合性）は訓練初期にしか成立せず、以降は WD が表現更新を安定化させる。  
- µP の効果は、実質的に「暗黙的ウォームアップ」として機能しており、強化された明示的ウォームアップで代替可能であると結論づける。
- 

---

#### 1. 背景と問題設定
大規模ニューラルネットの効率的学習では、小規模モデルで最適化された学習率をスケール転移する「学習率転移 (Learning Rate Transfer)」が重要である。  
µP はこの目的のために設計され、幅方向のスケーリングで表現変化 ∥ΔY∥ を安定化させる理論を提示してきた。  
しかし実務的には µP 単独では転移がうまくいかず、**独立した Weight Decay (Independent WD)** が必要であることが多くの報告で指摘されている。

<img width="676" height="766" alt="Image" src="https://github.com/user-attachments/assets/3f7af315-6b4b-4db9-bfa0-d230c85e6bfd" />

---

#### 2. 提案と主張
著者らは µP と Weight Decay を統一的に扱う「相対更新 (Relative Update)」の枠組みを提示し、  
- µP が想定するアラインメント比 (α_ΔW / α_W) は訓練初期のみ安定し、  
- 訓練後期には WD がこの不整合を打ち消して相対表現変化を安定化させる  
ことを実証した。  
特に **Independent WD** は µP のスケーリング則を打ち消し、訓練中期以降に更新量を均質化することで学習率転移を可能にしている。

---

#### 3. 新規性とこれまでの研究との違い
従来研究は µP の理論的正当性（無限幅近似、NTK との対比など）に焦点を当てていたが、  
本研究は **現実的スケールでの実証的解析** に基づき、µP の仮定がほとんどの訓練過程で破綻することを定量的に示した点が新しい。  
さらに、µP の効果を「学習率ウォームアップの一形態」として再定義し、  
実際には強化されたウォームアップスケジュールで同等の効果を再現できることを示した点で、理論的理解と実務的最適化を橋渡ししている。

---

#### 4. 実証と実験構成
LLaMA 言語モデル（幅128〜2048, 約1Bパラメータ）および ResNet を対象に大規模実験を実施。  
AdamW オプティマイザを用い、標準 WD・独立 WD・WDなしの3条件を比較した結果、次のような傾向が確認された：

- **Figure 1（p.1）**: µP は独立 WD と組み合わせた場合のみ、幅を変えても学習率転移が成功。  

<img width="566" height="284" alt="Image" src="https://github.com/user-attachments/assets/4cc3cb2f-0457-4c86-94f9-93e1c3af97fa" />

- **Figure 3（p.5）**: 標準 WD では相対表現変化が幅によって大きく異なるが、独立 WD はこれを一定に保つ。  

<img width="742" height="521" alt="Image" src="https://github.com/user-attachments/assets/9e94f64a-dca7-4df6-90d8-1f87ff7dbfbd" />

- **Figure 4（p.6）**: µP のアラインメント仮定（α_ΔW / α_W ∝ √C）は訓練初期しか成立せず、すぐ崩壊する。  

<img width="746" height="320" alt="Image" src="https://github.com/user-attachments/assets/d1e13059-c393-48c6-bb49-f57b7764e689" />

- **Figure 5（p.7）**: 独立 WD は学習初期にウォームアップのような効果を生み、相対更新を徐々に拡大させる。  

<img width="739" height="387" alt="Image" src="https://github.com/user-attachments/assets/af5c626c-0e72-4d53-80e5-4afd62aee5af" />

- **Figure 6（p.8）**: µP の代わりに指数関数的ウォームアップを導入しても同様の学習率転移が再現可能。

<img width="744" height="393" alt="Image" src="https://github.com/user-attachments/assets/205cbfe2-b538-420c-93a6-489537c79c63" />

これらにより、µP の効果は実際には **「暗黙的ウォームアップ」機構として機能している** ことが明らかになった。

---

#### 5. 貢献の検証
著者らは、相対更新解析（∥ΔW∥/∥W∥ と ∥ΔY∥/∥Y∥の対応）を通じて、  
独立 WD が µP の破綻した仮定を補正し、**訓練中期以降に表現変化を一定化**させていることを数値的に確認。  
また、µP を用いずとも指数型ウォームアップスケジュール (exp-increasing warmup) を導入すれば同等の結果が得られることを示し、  
実用的な最適化戦略として有効であることを実証した。

---

#### 6. 結論と示唆
- µP の理論は無限幅の仮定下では正しいが、有限幅・大バッチの実設定では早期に崩壊する。  
- 実際の学習率転移は **Independent Weight Decay** が主導しており、µP は主に初期学習の安定化を担う。  
- µP の効果はより強力な明示的ウォームアップで再現できる。  
- Matrix-level optimizer（例: Muon）ではアラインメントが一定に保たれるため、µP のような調整を不要にできる可能性がある。

---

### 図の説明（主要Figure）
- **Figure 1:** µP＋独立WDでのみLLaMAにおける学習率転移が成功。  
- **Figure 3:** 独立WDが幅間で相対表現変化を一定に保つことを可視化。  
- **Figure 4:** µPのアラインメント仮定（α_ΔW/α_W = √C）は訓練初期にしか成立しない。  
- **Figure 5:** µPの学習率スケーリングはウォームアップ的効果を持つことを示す。  
- **Figure 6:** 明示的な指数ウォームアップによりµPなしでも同等の性能が得られる。

---

### 総括
本研究は、「µP の実際的有効性は Weight Decay によって支えられている」という逆転的な発見を通じ、  
LLM トレーニングにおける学習率転移の理解を刷新した。  
µP を理論的支柱から実務的ツールへと再定義し、学習率設計・ウォームアップ戦略に新たな指針を与える貢献を果たしている。
